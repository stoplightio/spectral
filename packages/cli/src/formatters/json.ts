import { Formatter } from './types';

export const json: Formatter = results => {
  const outputJson = results.map(result => {
    return {
      code: result.code,
      path: result.path,
      message: result.message,
      severity: result.severity,
      range: result.range,
      source: result.source,
    };
  });
  return JSON.stringify(outputJson, null, '\t');
};
