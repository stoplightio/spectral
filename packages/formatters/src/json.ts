import { Formatter } from './types';

export const json: Formatter = results => {
  const outputJson = results.map(result => {
    let documentationUrlObject = {};
    if (result.documentationUrl) {
      documentationUrlObject = {
        documentationUrl: result.documentationUrl,
      };
    }
    return {
      code: result.code,
      path: result.path,
      message: result.message,
      severity: result.severity,
      range: result.range,
      source: result.source,
      ...documentationUrlObject,
    };
  });
  return JSON.stringify(outputJson, null, '\t');
};
