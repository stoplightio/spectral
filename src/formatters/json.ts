import { IRuleResult } from '../types';

export const json = (results: IRuleResult[]): string => {
  const outputJson = results.map(result => {
    return {
      code: result.code,
      path: result.path,
      message: result.message,
      severity: result.severity,
      range: result.range,
      source: result.source,
    };
  });
  return JSON.stringify(outputJson, null, '\t');
};
