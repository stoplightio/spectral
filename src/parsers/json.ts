import { getLocationForJsonPath, JsonParserResult, parseWithPointers, trapAccess } from '@stoplight/json';
import { IParser } from './types';

export const parseJson = (input: string) =>
  parseWithPointers(input, { ignoreDuplicateKeys: false, preserveKeyOrder: true });

export const Json: IParser<JsonParserResult<unknown>> = {
  parse: parseJson,
  getLocationForJsonPath,
  trapAccess,
};
