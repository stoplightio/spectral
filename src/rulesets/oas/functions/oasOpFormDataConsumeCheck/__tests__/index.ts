import { RuleType, Spectral } from '../../../../../index';
import { commonOasFunctions } from '../../../index';

import { rules } from '../../../index.json';

const ruleset = { functions: commonOasFunctions(), rules };

describe('oasOpFormDataConsumeCheck', () => {
  const s = new Spectral();
  s.addFunctions(ruleset.functions || {});
  s.addRules({
    'operation-formData-consume-check': Object.assign(ruleset.rules['operation-formData-consume-check'], {
      recommended: true,
      type: RuleType[ruleset.rules['operation-formData-consume-check'].type],
    }),
  });

  test('validate a correct object', async () => {
    const results = await s.run({
      paths: {
        '/path1': {
          get: {
            consumes: ['application/x-www-form-urlencoded', 'application/xml'],
            parameters: [{ in: 'formData', name: 'test' }],
          },
        },
      },
    });
    expect(results.length).toEqual(0);
  });

  test('return errors on different path operations same id', async () => {
    const results = await s.run({
      paths: {
        '/path1': {
          get: {
            consumes: ['application/xml'],
            parameters: [{ in: 'formData', name: 'test' }],
          },
        },
      },
    });

    expect(results).toMatchSnapshot();
  });
});
