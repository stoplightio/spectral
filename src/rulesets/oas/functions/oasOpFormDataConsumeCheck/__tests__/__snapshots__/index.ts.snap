// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oasOpFormDataConsumeCheck return errors on different path operations same id 1`] = `
Array [
  Object {
    "code": "operation-formData-consume-check",
    "message": "Operations with an \`in: formData\` parameter must include \`application/x-www-form-urlencoded\` or \`multipart/form-data\` in their \`consumes\` property.",
    "path": Array [
      "paths",
      "/path1",
      "get",
    ],
    "range": Object {
      "end": Object {
        "character": 26,
        "line": 10,
      },
      "start": Object {
        "character": 12,
        "line": 3,
      },
    },
    "severity": 1,
    "source": undefined,
    "summary": "Operations with an \`in: formData\` parameter must include \`application/x-www-form-urlencoded\` or \`multipart/form-data\` in their \`consumes\` property.",
  },
]
`;
